editor/utils/block-list.js

```diff
@@ -34,29 +43,89 @@ export function createInnerBlockList( uid, renderBlockMenu, showContextualToolba
 	if ( ! INNER_BLOCK_LIST_CACHE[ uid ] ) {
 		INNER_BLOCK_LIST_CACHE[ uid ] = [
 			// The component class:
-			class extends Component {
-				componentWillMount() {
-					INNER_BLOCK_LIST_CACHE[ uid ][ 1 ]++;
-				}
+			compose(
+				withSelect( ( select ) => {
+					const { getBlock, getBlockListSettings } = select( 'core/editor' );
+					return {
+						block: getBlock( uid ),
+						blockListSettings: getBlockListSettings( uid ),
+					};
+				} ),
+				withDispatch( ( dispatch ) => {
+					const { insertBlocks, updateBlockListSettings } = dispatch( 'core/editor' );
+					return {
+						insertBlocks( blocks ) {
+							dispatch( insertBlocks( blocks, undefined, uid ) );
+						},
+						updateNestedSettings( settings ) {
+							dispatch( updateBlockListSettings( uid, settings ) );
+						},
+					};
+				} ),
+			)(
```

Comment:
Minor: Would it be clearer if we move the HoC after the class declaration? and move it before the array assignment?

```
const MyClass = class Something {};
const MyClassContainer =compose(...)(myClass);
INNER_BLOCK_LIST_CACHE[ uid ] = [
  MyClassContainer, 0
]
```